### PR TITLE
feat(core,schemas): add phrases schema and GET /custom-phrases/:languageKey route

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

- Add `phrases` schema (LOG-4060)
- Add `GET /custom-phrases/:languageKey` to get the custom phrase translation of the specified language.

## Testing

image

Locally tested.

image

### CODE REVIEW

File: packages/core/src/routes/language.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Languages } from '@logto/schemas';
+
+import koaGuard from '@/middleware/koa-guard';
+import { findLanguageById } from '@/queries/language';
+
+import { AuthedRouter } from './types';
+
+export default function languageRoutes<T extends AuthedRouter>(router: T) {
```
Comment (simeng-li) [inline packages/core/src/routes/language.ts:8]: translations or i18n instead of language?

Comment (IceHe) [reply packages/core/src/routes/language.ts:8]: per offline discussion: we will rename related names to `localization`

File: packages/schemas/tables/phrases.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+create table phrases (
```
Comment (gao-sun) [inline packages/schemas/tables/phrases.sql:1]: better use `custom_phrases`?

Comment (IceHe) [reply packages/schemas/tables/phrases.sql]: good call

**Review by wangsijie — APPROVED** [review]
LGTM

File: packages/core/src/routes/init.ts [hunk 3]
```diff
@@ -39,6 +40,7 @@ const createRouters = (provider: Provider) => {
   logRoutes(managementRouter);
   roleRoutes(managementRouter);
   dashboardRoutes(managementRouter);
+  customPhraseRoutes(managementRouter);
```
Comment (charIeszhao) [inline packages/core/src/routes/init.ts:43]: Should this be coupled with our `AuthedRouter`? Or `AnonymousRouter` is good to go?

Comment (charIeszhao) [reply packages/core/src/routes/init.ts:43]: OK, I get it now, and this `customPhraseRoutes` will only be used in admin console. Will provide other anonymous routes for usages in mainflow ui.

**Review by charIeszhao — APPROVED** [review]

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 4]
```diff
@@ -155,3 +155,15 @@ export const adminConsoleConfigGuard = z.object({
...
+ * Phrases
+ */
+
+export type Translation = {
+  [key: string]: string | Translation;
+};
+
+export const translationGuard: z.ZodType<Translation> = z.lazy(() =>
+  z.record(z.string().or(translationGuard))
+);
```
Comment (darcyYe) [inline packages/schemas/src/foundations/jsonb-types.ts:169]: can we infer type Translation by translationGuard?

File: packages/core/src/routes/custom-phrase.test.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { CustomPhrase } from '@logto/schemas';
```
[the remark this answers is not in the document]
Comment (IceHe) [reply packages/core/src/routes/custom-phrase.test.ts:1]: Stay the same since it is unnecessary to test the Zod type guards.

**Pull request merged**